### PR TITLE
boost 1.73.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

---

previous upgrade, #48282

### CODE REVIEW

File: Formula/boost.rb [hunk 1]
```diff
@@ -1,22 +1,11 @@
 class Boost < Formula
   desc "Collection of portable C++ source libraries"
   homepage "https://www.boost.org/"
-  revision 2
   head "https://github.com/boostorg/boost.git"
 
   stable do
-    url "https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2"
-    mirror "https://dl.bintray.com/homebrew/mirror/boost_1_72_0.tar.bz2"
```
Comment (Bo98) [inline Formula/boost.rb:9]: Please keep the mirror. I'd also move the url and sha256 between `homepage` and `head` now that the patch is removed.

Comment (chenrui333) [reply Formula/boost.rb:9]: agree, I think the mirror got removed during the `brew bump-formula-pr` command process.

**Comment by Bo98** [comment]
CMake 3.17.2 is out with Boost 1.73 support: #53924

File: Formula/boost-python3.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 class BoostPython3 < Formula
   desc "C++ library for C++/Python3 interoperability"
   homepage "https://www.boost.org/"
-  url "https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2"
-  mirror "https://dl.bintray.com/homebrew/mirror/boost_1_72_0.tar.bz2"
-  sha256 "59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722"
+  url "https://dl.bintray.com/boostorg/release/1.73.0/source/boost_1_73_0.tar.bz2"
+  mirror "https://dl.bintray.com/homebrew/mirror/boost_1_73_0.tar.bz2"
+  sha256 "4eb3b8d442b426dc35346235c8733b5ae35ba431690e38c6a8263dce9fcbb402"
   revision 1
```
Comment (Bo98) [inline Formula/boost-python3.rb:7]:
```suggestion
```

**Comment by Bo98** [comment]
* ⚠️ `boost-bcp` does not compile on Catalina

---

* `freeling` needs a revision bump
* `libtorrent-rasterbar` has a test failure (except on High Sierra):

```
==> /usr/bin/clang++ -std=c++11 -I/usr/local/opt/boost/include/boost -L/usr/local/Cellar/libtorrent-rasterbar/1.2.6/lib -ltorrent-rasterbar -L/usr/local/opt/boost/lib -lboost_system -framework SystemConfiguration -framework CoreFoundation /usr/local/Cellar/libtorrent-rasterbar/1.2.6/libexec/examples/make_torrent.cpp -o test
In file included from /usr/local/Cellar/libtorrent-rasterbar/1.2.6/libexec/examples/make_torrent.cpp:33:
In file included from /usr/local/include/libtorrent/entry.hpp:61:
/usr/local/include/libtorrent/config.hpp:456:11: error: no type named 'size_t' in namespace 'std'; did you mean simply 'size_t'?
constexpr std::size_t TORRENT_READ_HANDLER_MAX_SIZE = 432;
          ^~~~~~~~~~~
          size_t
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/_types/_size_t.h:31:32: note: 'size_t' declared here
typedef __darwin_size_t        size_t;
```

* :x: `gnuradio` does not compile with Boost 1.73
* :x: `monero` does not compile with Boost 1.73
* :x: `widelands` does not compile with Boost 1.73

**Comment by selsta** [comment]
> ❌ monero does not compile with Boost 1.73

The latest monero release (v0.16.0.0) is compatible with boost 1.73.

**Comment by chenrui333** [comment]
Going to kick off another run to see where we are.

File: Formula/bitcoin.rb [hunk 3]
```diff
@@ -1,8 +1,9 @@
 class Bitcoin < Formula
   desc "Decentralized, peer to peer payment network"
   homepage "https://bitcoin.org/"
-  url "https://bitcoin.org/bin/bitcoin-core-0.20.0/bitcoin-0.20.0.tar.gz"
-  sha256 "ec5a2358ee868d845115dc4fc3ed631ff063c57d5e0a713562d083c5c45efb28"
+  url "https://bitcoin.org/bin/bitcoin-core-0.19.1/bitcoin-0.19.1.tar.gz"
+  sha256 "f2591d555b8e8c2e1bd780e40d53a91e165d8b3c7e0391ae2d24a0c0f23a7cc0"
```
Comment (chenrui333) [inline Formula/bitcoin.rb:5]: will redo this.

**Comment by chenrui333** [comment]
cancelled the boost run as too many things going on (gcc, sqlite, and python3.7 upgrade)

**Comment by SMillerDev** [comment]
Rebased and pushed. 🤞

**Comment by chenrui333** [comment]
Current status:

on 10.15
Error: 8 failed steps!
brew test --verbose freeling
brew test --verbose lgogdownloader
brew test --verbose libpulsar
brew test --verbose ncmpcpp
brew test --verbose widelands
brew test --verbose adios2
brew install --verbose --build-bottle boost-bcp
brew install --verbose --build-bottle gnuradio

on 10.14
Error: 7 failed steps!
brew test --verbose freeling
brew test --verbose lgogdownloader
brew test --verbose libpulsar
brew test --verbose ncmpcpp
brew test --verbose widelands
brew test --verbose adios2
brew install --verbose --build-bottle gnuradio
##[error]Process completed with exit code 1.

on 10.13
Error: 7 failed steps!
brew test --verbose freeling
brew test --verbose lgogdownloader
brew test --verbose libpulsar
brew test --verbose ncmpcpp
brew test --verbose widelands
brew test --verbose adios2
brew install --verbose --build-bottle gnuradio

**Comment by SMillerDev** [comment]
Rebasing, if it's just widelands and gnuradio I think we should just merge

**Comment by chenrui333** [comment]
> Rebasing, if it's just widelands and gnuradio I think we should just merge

Sounds good!

**Comment by SeekingMeaning** [comment]
10.15:

```
Error: 3 failed steps!
brew test --verbose adios2
brew install --verbose --build-bottle gnuradio
brew install --verbose --build-bottle widelands
```

@dtrodrigues has a patch for `libfabric` that fixes `adios2`, see #58349

**Comment by dtrodrigues** [comment]
libfabric patch separated into https://github.com/Homebrew/homebrew-core/pull/58437

**Comment by SMillerDev** [comment]
Just gnuradio left. Can someone merge this when it's ready?

**Comment by chenrui333** [comment]
> Just gnuradio left. Can someone merge this when it's ready?

yeah, I will take on it.

**Comment by chenrui333** [comment]
looks like we have lots of new style checks right now :(

**Comment by chenrui333** [comment]
waiting for 10.14 build to finish.

**Comment by SeekingMeaning** [comment]
This looks like it almost about to be merged, but I thought should mention that I have a fix for `gnuradio`: #58751

**Comment by SMillerDev** [comment]
It's been going for a day. Restarting the runners.

**Comment by Bo98** [comment]
>  I thought should mention that I have a fix for `gnuradio`: #58751

The problem isn't really just about shims, it's that the version that we ship is outright incompatible with Boost 1.73. We would have to backport a patch ourselves since upstream haven't provided a 3.8.x version suitable for distribution in Homebrew yet.

**Comment by jonchang** [comment]
I'm doing a manual merge which will take a few hours, so please don't touch this pull request in the meantime.

**Comment by chenrui333** [comment]
> I'm doing a manual merge which will take a few hours, so please don't touch this pull request in the meantime.

Thanks!

**Comment by jonchang** [comment]
Merged in e85103271f5d30f7c7802c3cc022c9880769bdef.

**Pull request closed without merging**